espree.js

```diff
@@ -3857,6 +3858,15 @@ function parseParams(firstRestricted) {
                     message = Messages.StrictParamDupe;
                 }
             }
+            if (match("=")) {
+                if (!allowDefaultParams) {
+                    throwUnexpected(lookahead);
+                }
+                lex();
+                def = parseAssignmentExpression();
+                ++defaultCount;
+            }
+            defaults.push(def);
```

Comment:
Answer is yes it needs to be there, so that the `defaults` array can be filled with `null` values in the case that there isn't a default for one of the arguments. (See latest update)